app/src/main/java/de/dhbw/corona_world_app/ui/tools/StatisticCallRecyclerViewFragment.java

```diff
@@ -128,4 +127,23 @@ public void onStop() {
     public abstract ShowStatisticInterface getShowStatisticInterface();
 
     public abstract void initViewModelData(StatisticCallViewModel statisticCallViewModel);
+
+    protected void tryRepairingData(){
+        ErrorDialog.createBasicErrorDialog(getContext(), "Your Data seems to be corrupt", "We were not able to save new Data, we will try to fix this Problem", (dialog, which) -> {
+            //TODO implement check to see if Data can be recovered
+            boolean canBeRecovered=false;
+            if(canBeRecovered){
+                //recover Data
+            }else{
+                ErrorDialog.createBasicErrorDialog(getContext(), "We were not able to recover your Data", "Your History and your Favourites must be deleted for this app to function properly, we are so sorry", (dialog1, which1) -> {
+                    try {
+                        statisticCallViewModel.deleteAllItems().get();
+                    } catch (ExecutionException | InterruptedException e1) {
+                        Log.e(this.getClass().getName()+"|"+getDataType(),"Not able to delete corrupt Data", e1);
+                        ErrorDialog.createBasicErrorDialog(getContext(),"There has been an error deleting your Data","Something has gone terribly wrong, please reinstall the app and try again",null);
```

Comment:
Look above. "Reinstall the app" should not be a message we want to give the user...